### DISCUSSION TITLE
How to transfer Cloudlog to another server?

### DISCUSSION THREAD

**Discussion by MikleBas** [magicbug/Cloudlog#2758] [Q&A] [answered]
I use a rather old version of this wonderful magazine, but attempts to upgrade constantly run into git errors, to be honest, I’m not good at it and it’s easier for me to transfer the entire magazine with all the settings to another VPS
How and what should I do to save the connection log and transfer all my settings?
Thank you

**Comment by HB9HIL** [comment]
Hello @MikleBas 

Can you tell me which Server OS do you use? Ubuntu? Debian?

So then I can write you a detailed Step by Step how you can transfer your Cloudlog Installation. This will probably also help others :)

**Reply by MikleBas** [reply]
Hello!
Thanks for responding :)
Now there is a clean Ubuntu 22.04.3 LTS
although I can install Debian, they are both good and I more or less know them

**Comment by HB9HIL** [comment, marked as answer]
Hi @MikleBas 

I post this tutorial here you can follow IF you don't have any special configurations AND you know what you are doing on a Linux system. If you aren't experienced with working on the commandline you should E-Mail me (Mail is on QRZ.com) and I can help you directly. If you follow this Step by Step Guide I won't be responsible for data loss. You have to edit some comments by yourself. This is not Copy Paste! This is just a guide.

if you install Ubuntu or Debian doesn't matter. I personally like Debian because it's a bit thinner and from my point of view a bit more stable. I would suggest Debian 12 "Bookworm"

The following Step's do not guarantee everything working ok. In this steps all commands are performed as `root`! Do this on your own risk. **You need to know how you can handle problems.** This tutorial is no installation step-by-step. You also have to set up the webserver by yourself.

First make sure you act as `root`
```
sudo su
```

## Step 1 - Backup's

You first should create a proper backup of the data and the SQL database:
For this we create a backup folder and copy all the files from webroot.

```
mkdir -p /root/cloudlog-backup/webfolder
cp -r /var/www/html /root/cloudlog-backup/webfolder
```

> Hint: Maybe you have to change `/var/www/html` to your particular installation path.

Then you also should make a backup of your database:

```
mysqldump cloudlog > /root/cloudlog-backup/cloudlog-database-backup.sql
```

> Hint: Here aswell. If the name of your cloudlog database is not `cloudlog` you have to edit this command.

To make it easier to copy all files we create a compressed tar file

```
tar -czvf /root/cloudlog-backup.tar.gz -C /root cloudlog-backup
```

The file `cloudlog-backup.tar.gz` is now a full backup of your Cloudlog Installation. Safe this one on a safe place.

## Step 2 - New Machine

In the next Step install the latest Debian (or Ubuntu) on the new Machine and install Cloudlog as described in the [Wiki](https://github.com/magicbug/Cloudlog/wiki/Installation). **Only perform the Step's 1 to 4!!!** Don't forget the [Cronjobs](https://github.com/magicbug/Cloudlog/wiki/Recommended-Cron-Jobs)!

Make sure you use the same Database Name, Database User and Database Password as in your old installation. Otherwise you have to adjust the credentials in the new installation manually. You can find the set credentials in the file `application/config/database.php`.

Only perform the Step's 1 to 4. You don't need to run the installation wizard in this case.

## Step 3 - Modify the Installation

Now we modify the new installation. For this we decompress the `cloudlog-backup.tar.gz` to the path `/tmp`. This folder will be deleted after some time automatically because we don't need it anymore after we modified everything. How you copy the file you have to figure out by yourself. I would suggest to copy the file with SFTP to the root folder (`/root`)

We now need to decompress this file

```
cd
tar -xzvf cloudlog-backup.tar.gz -C /tmp
```
If this is done we can start to replace some files. I use the installation path `/var/www/html` as described in the Wiki. Maybe you have to adjust this path to your particular installation path.

```
rm -r /var/www/html/install
```
> This removes the Installation Wizard since we don't need this one in this case

```
cp /tmp/cloudlog-backup/webfolder/html/application/config/config.php /var/www/html/application/config/config.php
```
> This copies the `config.php` to the new installation. ATTENTION: Compare the copied `config.php` with the `config.sample.php`. If you run a really old version Cloudlog you may need to add some config variables.

```
cp /tmp/cloudlog-backup/webfolder/html/application/config/database.php /var/www/html/application/config/database.php
```
> This copies the Database Configuration to the new installation. If you created the Database during the new installation with different credentials then your old installation you have to edit this file manually. 

```
rsync -a /tmp/cloudlog-backup/webfolder/html/uploads /var/www/html/uploads
rsync -a /tmp/cloudlog-backup/webfolder/html/updates /var/www/html/updates
rsync -a /tmp/cloudlog-backup/webfolder/html/images/eqsl_card_images /var/www/html/images/eqsl_card_images
rsync -a /tmp/cloudlog-backup/webfolder/html/assets/qslcard /var/www/html/assets/qslcard
```
With this commands we sync uploads, qslcards and data files.

After we copied all those files we want to make sure the permissions in the installation folder are set correctly

```
chown -R root:www-data /var/www/html/application/config/
chown -R root:www-data /var/www/html/application/logs/
chown -R root:www-data /var/www/html/assets/qslcard/
chown -R root:www-data /var/www/html/backup/
chown -R root:www-data /var/www/html/updates/
chown -R root:www-data /var/www/html/uploads/
chown -R root:www-data /var/www/html/images/eqsl_card_images/
chown -R root:www-data /var/www/html/assets/json/
chmod -R g+rw /var/www/html/application/config/
chmod -R g+rw /var/www/html/application/logs/
chmod -R g+rw /var/www/html/assets/qslcard/
chmod -R g+rw /var/www/html/backup/
chmod -R g+rw /var/www/html/updates/
chmod -R g+rw /var/www/html/uploads/
chmod -R g+rw /var/www/html/images/eqsl_card_images/
chmod -R g+rw /var/www/html/assets/json/
```

## Step 4 - SQL Database

During Step 2 of this Tutorial you created a MySQL Database while following the installation wiki. We will now restore the SQL Backup we did on the old machine and populate this data into the new database.

First we have to create the tables (basic redump)
```
mysqldump cloudlog < /tmp/cloudlog-backup/cloudlog-database-backup.sql
```
Then we hop into mysql to populate all the necessary data
```
mysql
```
Perform this commands in the mysql commandline
```
USE cloudlog;
SOURCE /tmp/cloudlog-backup/cloudlog-database-backup.sql;
```
## Step 5 - Feedback

Restart the webserver. I used apache2.
```
systemctl restart apache2
```
Congrats! You should successfully transfered your Cloudlog-instance to a new server. Please test EVERYTHING and report how you were able to follow this small tutorial. 

Vy 73 de HB9HIL

**Reply by MikleBas** [reply]
Hello!
Thank you very much for your work, this is very valuable material!
Whenever I have free time, I will move to another server according to your instructions; in any case, I will report the result here.
